clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -62,6 +64,7 @@ class Util {
   /// Checks whether given clang type is a full specialization of the SYCL
   /// accessor class.
   static bool isSyclAccessorType(const QualType &Ty);
+  static bool isSyclAccessorType(const CXXRecordDecl *RecTy);
```

Comment:
Why are these overloads necessary?  Is it not possible to just make the uses do RecordDecl->getType()?

Reply:
This is another interface for convenience. Plus this avoids unnecessary dereferences, dyn_cast and branch:
```
bool Util::isSyclAccessorType(const QualType &Ty) {
  if (const CXXRecordDecl *RecTy = Ty->getAsCXXRecordDecl())
    return isSyclAccessorType(RecTy);
  return false;
}
```

Reply:
I think I'd prefer just having people write ->getType instead of this.  You end up repeating all the record-decl detection code 1/2 dozen times.

Reply:
This is matter of taste. If you insist I can change - ?

Reply:
I think your change reduces readability, and is otherwise an unrelated change.  @Fznamznon , thoughts?

Reply:
I don't have strong preference regarding this, but for me it seems that there become a bit more code than it was before, so I don't mind reverting it.